plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/N4JSTypeValidator.xtend

```diff
@@ -576,137 +641,132 @@ class N4JSTypeValidator extends AbstractN4JSDeclarativeValidator {
-	}
-
-	def private void checkComposedTypeRefContainsNoAny(ComposedTypeRef ctr, String msg, String issueCode, boolean soleVoidAllowesAny) {
-		val G = ctr.newRuleEnvironment;
-		val anyType = G.anyType;
-		val voidType = G.voidType;
-		val EList<TypeRef> typeRefs = ctr.getTypeRefs();
-		val Iterable<TypeRef> anyTypeRefs = typeRefs.filter[it.getDeclaredType()===anyType]; // identity check on typeRefs is OK here
-
-		var boolean dontShowWarning = false;
-		if (soleVoidAllowesAny) {
-			val boolean containsVoid = typeRefs.exists[it.getDeclaredType()===voidType]; // identity check on typeRefs is OK here
-			dontShowWarning = containsVoid && anyTypeRefs.size() == 1;
-		}
-
-		if (!dontShowWarning) {
-			for (TypeRef anyTR : anyTypeRefs) {
-				addIssue(msg, anyTR, issueCode);
+			/**
+			 * This validates a warning in chapter 4.10.2:<br/>
+			 * <i>The use of the any type in an intersection type produces a warning.</i>
+			 */
+			@Check
+			def void checkIntersectionTypeContainsNoAny(IntersectionTypeExpression ite) {
+				checkComposedTypeRefContainsNoAny(ite, messageForINTER_ANY_USED, INTER_ANY_USED, false);
 			}
-		}
-	}
 
+			def private void checkComposedTypeRefContainsNoAny(ComposedTypeRef ctr, String msg, String issueCode,
+				boolean soleVoidAllowesAny) {
+				val G = ctr.newRuleEnvironment;
+				val anyType = G.anyType;
+				val voidType = G.voidType;
+				val EList<TypeRef> typeRefs = ctr.getTypeRefs();
+				val Iterable<TypeRef> anyTypeRefs = typeRefs.filter[it.getDeclaredType() === anyType]; // identity check on typeRefs is OK here
+				var boolean dontShowWarning = false;
+				if (soleVoidAllowesAny) {
+					val boolean containsVoid = typeRefs.exists[it.getDeclaredType() === voidType]; // identity check on typeRefs is OK here
+					dontShowWarning = containsVoid && anyTypeRefs.size() == 1;
+				}
 
-	/**
-	 * This validates a warning in chapter 4.10.1:<br/>
-	 * <i>The use of unnecessary subtypes in union types produces a warning.</i>
-	 */
-	@Check
-	def void checkUnionHasUnnecessarySubtype(UnionTypeExpression ute) {
-		val List<TypeRef> tRefs = extractNonStructTypeRefs(ute);
-		val G = ute.newRuleEnvironment;
-		val anyType = G.anyType;
-		tRefs.removeIf[it.getDeclaredType === anyType]; // identity check on typeRefs is OK here
-
-		if (tRefs.size() > 1) {
-			val List<TypeRef> intersectionTR = tsh.getSuperTypesOnly(G, tRefs);
-			tRefs.removeAll(intersectionTR);
-
-			for (TypeRef tClassR : tRefs) {
-				val message = messageForUNI_REDUNDANT_SUBTYPE;
-				addIssue(message, tClassR, UNI_REDUNDANT_SUBTYPE);
+				if (!dontShowWarning) {
+					for (TypeRef anyTR : anyTypeRefs) {
+						addIssue(msg, anyTR, issueCode);
+					}
+				}
 			}
-		}
-	}
 
+			/**
+			 * This validates a warning in chapter 4.10.1:<br/>
+			 * <i>The use of unnecessary subtypes in union types produces a warning.</i>
+			 */
+			@Check
+			def void checkUnionHasUnnecessarySubtype(UnionTypeExpression ute) {
```

Comment:
bogus indentation due to accidental auto format; please undo

Reply:
fixed (and Moritz has to pay 1€ to the piggybank ;-) )